helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -527,8 +540,11 @@ public void invoke(NotificationContext changeContext) throws Exception {
       _zkClient.unsubscribeChildChanges(path, this);
     }
 
-    // List of children could be empty, but won't be null.
-    return _zkClient.getChildren(path);
+    try {
+      return _zkClient.getChildren(path);
+    } catch (ZkNoNodeException e) {
+      return null;
+    }
```

Comment:
This changes the behavior, right? Can you elaborate what kind of scenario you will see this?

Reply:
This change was because you reminded me about the instances without the task path. :) 

For instance without TASKCURRENTSTATES path, the callback handler is still created, however, during reset(), subscribeChildChange() will be invoked and they will reach the above path. Since the path doesn't exist, this line causes a NoNodeException that's only caught on the outside by reset(). This causes some log pollution. 

I synced with @pkuwm and we determined that this is unexpected and should be fixed.

Reply:
My concern is if original behavior of other types will be changed as well. The exception we will see can be changed from NoNodeException to NullPointerException if you just return null for it.

Reply:
All callers of this function (2 occasions) handle null cases. I believe this is the best solution when `path` doesn't exist, unless there's a better idea.